ls1intum/Artemis
# `File upload exercises`: Allow import of file upload exercises

### Checklist
#### General


#### Server
#### Client

### Motivation and Context
File Upload Exercises have been the only exercise type for which the import functionality was missing.
### Description

This adds two new REST endpoints in the `FileUploadExerciseResource`: ` POST file-upload-exercises/import/:id` for the actual import and `GET file-upload-exercises` to get all file-upload exercises are user has access to.
On the client side the `file-upload-exercise-update-component` is updated and the `paging-service` is introduced for file-upload exercises as well as an import method in the `file-upload-exercise-service`

### Steps for Testing

Prerequisites:
- 1 Editor or more privileged user
- 1 Course with a File Upload Exercise 
- 1 Course where the exercise is not yet part of.

1. Log in to Artemis
2. Navigate to Course Administration
3. Go to the course where you want to import the exercise
4. Scroll down to File Upload Exercises and click Import File Upload Exercise
5. Search for the name of the exercise (or directly select it if it's shown on the first page)
6. Fill some fields on the page that opens and click Save.
7. The exercise should now be imported into the course

#### Exam Mode Testing
Doesn't affect exam mode


### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage





### Screenshots


![file-upload-exercise-import](https://user-images.githubusercontent.com/84102468/215044990-dcdf9a80-e1dc-48c6-914c-21ea331d7b33.gif)
